File: sdk/python/edufish/exceptions.py
"""EDUFISH SDK exceptions."""


class EduFishError(Exception):
    """Base exception for all SDK errors."""

    def __init__(self, message: str, code: str = "UNKNOWN", status: int = 0):
        self.code = code
        self.status = status
        super().__init__(message)


class AuthenticationError(EduFishError):
    """Raised when API key is missing or invalid."""

    def __init__(self, message: str = "Invalid or missing API key"):
        super().__init__(message, code="UNAUTHORIZED", status=401)


class NotFoundError(EduFishError):
    """Raised when a requested resource does not exist."""

    def __init__(self, message: str = "Resource not found"):
        super().__init__(message, code="NOT_FOUND", status=404)


class ValidationError(EduFishError):
    """Raised when request data fails validation."""

    def __init__(self, message: str = "Validation failed"):
        super().__init__(message, code="UNPROCESSABLE", status=422)


class ServerError(EduFishError):
    """Raised when the engine returns a 5xx error."""

    def __init__(self, message: str = "Internal server error"):
        super().__init__(message, code="INTERNAL_ERROR", status=500)

File: sdk/js/src/exceptions.ts
export class EduFishError extends Error {
  code: string;
  status: number;

  constructor(message: string, code = "UNKNOWN", status = 0) {
    super(message);
    this.name = "EduFishError";
    this.code = code;
    this.status = status;
  }
}

export class AuthenticationError extends EduFishError {
  constructor(message = "Invalid or missing API key") {
    super(message, "UNAUTHORIZED", 401);
    this.name = "AuthenticationError";
  }
}

export class NotFoundError extends EduFishError {
  constructor(message = "Resource not found") {
    super(message, "NOT_FOUND", 404);
    this.name = "NotFoundError";
  }
}

export class ValidationError extends EduFishError {
  constructor(message = "Validation failed") {
    super(message, "UNPROCESSABLE", 422);
    this.name = "ValidationError";
  }
}

export class ServerError extends EduFishError {
  constructor(message = "Internal server error") {
    super(message, "INTERNAL_ERROR", 500);
    this.name = "ServerError";
  }
}
